pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,272 @@ type StorageVolumeList struct {
+type Affinity struct {
+
+	// Describes volume affinity scheduling rules
+	// +optional
+	VolumeAffinity []VolumeAffinityTerm `json:"volumeAffinity,omitempty"`
+
+	// Describes volume anti-affinity scheduling rules
+	// +optional
+	VolumeAntiAffinity []VolumeAffinityTerm `json:"volumeAntiAffinity,omitempty"`
+
+	// Describes cohort affinity scheduling rules
+	// +optional
+	CohortAffinity []metav1.LabelSelector `json:"cohortAffinity,omitempty"`
+
+	// Describes cohort anti-affinity scheduling rules
+	// +optional
+	CohortAntiAffinity []metav1.LabelSelector `json:"cohortAntiAffinity,omitempty"`
+
+	// TopologySpreadConstraint specifies how to spread matching volumes among the given topology.
+	// +optional
+	TopologySpreadConstraint []v1.TopologySpreadConstraint `json:"topologySpreadConstraint,omitempty"`
+}
+
+type VolumeAffinityTerm struct {
+	TopologyKey string `json:"topologyKey,omitempty"`
+
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+
+// StorageVolumeStatus defines the observed state of StorageVolume
+// +k8s:openapi-gen=true
+type StorageVolumeStatus struct {
+	// Phase defines the state of the volume
+	Phase StorageVolumePhase `json:"phase"`
+
+	// Capacity stores total, used and available size
+	// +nullable
+	Capacity Capacity `json:"capacity,omitempty"`
+
+	// NVMESpec defines the nvme spec i.e nql, targetIP and status
+	// +nullable
+	NVMESpec []NVMESpec `json:"NVMESpec,omitempty"`
+}
+
+// Capacity defines total, used and available size
+type Capacity struct {
+	// Total represents total size of volume
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Used represents used size of the volume
+	Used resource.Quantity `json:"used,omitempty"`
+
+	// Available represents available size of the volume
+	Available resource.Quantity `json:"available,omitempty"`
+
+	// Error denotes the error occurred during provisioning a volume.
+	// Error field should only be set when State becomes Failed.
+	Error *StorageVolumeError `json:"error,omitempty"`
+}
+
+// NVMESpec represents nvme spec for the volume
+type NVMESpec struct {
+
+	// Nqn represents target nvme Qualified Name.combination of nodeBase
+	Nqn string `json:"nqn,omitempty"`
+
+	// TargetIP IP of the NVME target
+	TargetIP string `json:"targetIP,omitempty"`
+
+	// Status of the nvme target
+	Status string `json:"status,omitempty"`
+}
+
+// StorageVolumePhase represents the current phase of StorageVolume.
+type StorageVolumePhase string
+
+const (
+	// StorageVolumePhasePending indicates that the StorageVolume is still waiting for
+	// the StorageVolume to be created and bound
+	StorageVolumePhasePending StorageVolumePhase = "Pending"
+
+	// StorageVolumePhaseFailed indicates that the StorageVolume provisioning
+	// has failed
+	StorageVolumePhaseFailed StorageVolumePhase = "Failed"
+
+	// StorageVolumePhaseReady indicates that the StorageVolume provisioning
+	// has Created
+	StorageVolumePhaseReady StorageVolumePhase = "Ready"
```

Comment:
what is ready state?
is there any difference between ready and healthy?

Reply:
Ready state is not required. Will remove it.